src/main/java/edu/harvard/iq/dataverse/DataversePage.java

```diff
@@ -562,48 +563,18 @@ public void updateOptionsRadio(Long mdbId, Long dsftId) {
 
     public String save() {
         List<DataverseFieldTypeInputLevel> listDFTIL = new ArrayList<>();
-        if (editMode != null && editMode.equals(EditMode.INFO)) {
+        if (editMode != null && (editMode.equals(EditMode.INFO) || editMode.equals(EditMode.CREATE))) {
+            List<MetadataBlock> selectedMetadataBlocks = getSelectedMetadataBlocks();
 
-            List<MetadataBlock> selectedBlocks = new ArrayList<>();
-            if (dataverse.isMetadataBlockRoot()) {
-                dataverse.getMetadataBlocks().clear();
-            }
-
-            for (MetadataBlock mdb : this.allMetadataBlocks) {
-                if (dataverse.isMetadataBlockRoot() && (mdb.isSelected() || mdb.isRequired())) {
-                    selectedBlocks.add(mdb);
-                    for (DatasetFieldType dsft : mdb.getDatasetFieldTypes()) {
-                        if (dsft.isRequiredDV() && !dsft.isRequired()
-                                && ((!dsft.isHasParent() && dsft.isInclude())
-                                || (dsft.isHasParent() && dsft.getParentDatasetFieldType().isInclude()))) {
-                            DataverseFieldTypeInputLevel dftil = new DataverseFieldTypeInputLevel();
-                            dftil.setDatasetFieldType(dsft);
-                            dftil.setDataverse(dataverse);
-                            dftil.setRequired(true);
-                            dftil.setInclude(true);
-                            listDFTIL.add(dftil);
-                        }
-                        if ((!dsft.isHasParent() && !dsft.isInclude())
-                                || (dsft.isHasParent() && !dsft.getParentDatasetFieldType().isInclude())) {
-                            DataverseFieldTypeInputLevel dftil = new DataverseFieldTypeInputLevel();
-                            dftil.setDatasetFieldType(dsft);
-                            dftil.setDataverse(dataverse);
-                            dftil.setRequired(false);
-                            dftil.setInclude(false);
-                            listDFTIL.add(dftil);
-                        }
-                    }
-                }
-            }
+            if (!selectedMetadataBlocks.isEmpty()) {
+                dataverse.setMetadataBlocks(selectedMetadataBlocks);
 
-            if (!selectedBlocks.isEmpty()) {
-                dataverse.setMetadataBlocks(selectedBlocks);
+                listDFTIL = getSelectedMetadataFields(selectedMetadataBlocks);
```

Comment:
Maybe something along the lines? :
```
if (dataverse.isMetadataBlockRoot()) {
    List<MetadataBlock> selectedMetadataBlocks = getSelectedMetadataBlocks();
    
    dataverse.setMetadataBlocks(selectedMetadataBlocks);
    listDFTIL = getSelectedMetadataFields(selectedMetadataBlocks);
}
```
And remove earlier `List<MetadataBlock> selectedMetadataBlocks = getSelectedMetadataBlocks();`
And in `getSelectedMetadataBlocks` remove all calls to `dataverse.isMetadataBlockRoot()`

Reply:
Fair enough but `dataverse.getMetadataBlocks().clear();` has to stay, otherwise there will be another Null exception, but understanding what is going on, is task for another few hours since in the 'getSelectedMetadataBlocks()' I'm not even looping through  'dataverse.getMetadataBlocks()'.